terraform/modules/queue/main.tf

```diff
@@ -47,6 +47,7 @@ data "aws_iam_policy_document" "sns_send_message" {
       values   = [var.sns_topic_arn]
     }
   }
+  source_policy_documents = var.policy_documents
```

Comment:
The issue with just adding source_policy_documents here is that the policy document will only be created if the sns_topic_arn variable is set.

We could remove the count conditional (here and in aws_sqs_queue_policy below) and associate the policy document regardless of whether sns_topic_arn is set, but that might break existing usage of the queue module by creating and associating a policy where that hadn't been done before.

I believe it will be safest to expand the count conditionals in these resources to include the source_policy_documents variable.